### ISSUE TITLE
Solar Issues in other Dimensions

### ISSUE THREAD

**Issue by NillerMedDild** [mekanica/mekanica#46]
#### Issue description:
Solar Generators do not work on a Space Station (Dimension ID -2) while it's raining in the overworld. Tested with Mekanism Solar Neutron Activators

#### Steps to reproduce:
1. Make an instance with Mekanism, Advanced Rocketry, JEI and preferably Just Enough Dimensions for easy teleporting, and Extra Utilities 2 for the Angel Block.
2. Toggle rain on in the overworld
3. Travel to dimension -2 (command `/tpj -2`)
4. Place an angel block, place Solar Neutron Activator on it
5. Observe the Solar Neutron Activator not working

#### Version (make sure you are on the latest version before reporting):
**Forge:** 1.12.2-14.23.5.2781
**Mekanism:** MekanismDz-1.12.2-9.5.0b3
**Advanced Rocketry:** AdvancedRocketry-1.12.2-1.5.0-132-universal

**Comment by dizzyd** [comment]
Excellent bug! Thank you for the concise reproduction steps. We'll look into getting this fixed for next week's release.

**Comment by NillerMedDild** [comment]
Thank you :D

**Comment by pupnewfster** [comment]
Haven't really looked into this yet, but my guess is that they may not be returning the correct value for `World#isRaining()` for their worlds. Needs to be looked into further though, because maybe that method only is meant to describe the overworld or things.

**Comment by dizzyd** [comment]
I've got a branch going that will address this issue. Our solar generation should now be both more efficient and realistic. It takes into account sun brightness, temperature and humidity and whether or not the chunk in which the generator is placed can experience rain. All that said, the end result is that the solars generate less power; details here: https://docs.google.com/spreadsheets/d/1hkx1R6KtNmyPf8_Xih1TnM5A3M_hqEyBYm8HHYzut2s/edit?usp=sharing

However, there is still a bug in Advanced Rocketry where they vary sun brightness depending on rain, even if the biome in question can't possibly have rain - I've added a note on their ticket.
